### ISSUE TITLE
AO Link: Major usability improvements

### ISSUE THREAD

**Issue by WendyStarfall** [OpenCollar/opencollar#984]
**Version:** `160528.3`

~~We receive the following errors when attempting to install this utility in AOs that are rezzed on the ground:~~

```
llGetAnimationOverride error:

neither PERMISSION_OVERRIDE_ANIMATIONS or PERMISSION_TRIGGER_ANIMATION permissions are set
```

The error above has already been ogre-fixed with 8e3b7ec3f9ab6e8a5c731474d1ee3adba71f637f. This issue is all about improvements now. 🎈 

I've had a hard time with the `oc_accessory_aolink` in general when trying to help users. That's nothing new per se but I witnessed once again how hard it is for some users to manually install a script.

I came to the conclusion that it is a bad idea to further supply this script as embedded utility with collars.

We have to provide it either as a dedicated product with more than sufficient documentation, or we simply link to the most recent RAW source of the script here on GitHub. I'd prefer the latter because in that case the user will always go through at least a bit of documentation before attempting to install the script.

Picking up at the latter case also means that users who already fail at installing would naturally require assistance to copy, paste and compile the source as well. This is healthy because it encourages other members in the community to help. Regardless how well we document things, this is and will be something where users need dedicated help from a human, which will ultimately lead to greater feedback, better tweaks and most importantly to a broader spectrum of compatible AOs.

Other challenges include features that allow to easily learn its version and to conveniently update it.

**Comment by WendyStarfall** [comment]
If anyone urgently needs an `oc_accessory_aolink` without error, I ogre-fixed it on the ~~alt.aolink branch~~.

**Comment by WendyStarfall** [comment]
Okay then, I merge this edit and will be doing a few more things, to break it down a real quick:
1. I'm going to make @GarvinTwine's method to turn AntiSlide off when using AO Link a general case. (This was at first only used in case the linked AO is a server side type but we don't really need that.)
2. I will necro part of Medea's idea to take controls for turning AOs on when movement buttons are pushed. In such an event the AO is turned on to override the walk so that we don't require AntiSlide.

What about AntiSlide then? It's still extremely important of course for everyone who uses client-side AOs!

**Comment by WendyStarfall** [comment]
Thanks @GarvinTwine for supervision and help (yep, I made mistakes with basic operators 🙊 oh dear )

What else needs to happen now?
- When the object's contents change, this script fires an LM that will be asking something like _"Is there anyone else like me? By the way I'm `g_sVersion`!"_
- If **yes**, the other script (also an ao link) will send another LM with _"Yes, me, I'm `g_sVersion`!"_.
- If this `g_sVersion` **is smaller** than the `g_sVersion` of the existing ao link, the existing ao link will **delete the new script**.
- If it **is larger** than the `g_sVersion` of the existing ao link, the existing ao link will **delete itself**.

**Comment by GarvinTwine** [comment]
OK, so in #987 I changed some things:

First I removed the "fake" ZHAO menu which is simply very experimental and might cause more issues than actual advantages.
Kept the working Oracul menu call, maybe other ZHAO based but adjusted AOs also have commands that can open the menu (in the default ZHAO there simply is none). If we know them, we can add them as well.
Then I added a version check versus other aolink scripts:
Scripts which the new version and later will check via LMs which is the higher version and remove the older one, silently! In case there is no script that replies to the LMs we check for scripts which have either "aolink" or "ao link" in their name and remove those, silently.
To be more aware that the aolink script "talks to you" I added a Say() function so notifies from the script gets send by the scipt's name.
New commands added:
`/88 rm aolink` will remove the script without further confirmation
`/88 debug on|off` will enable a debug mode:
All LinkMessages in the AO will be debugged with a llSay(0,"Debug: iNum= Message="), to be able to find possible new settings to add for other AOs without the need to use an addition debug script.

So far so good, I hope, now some notifies need for sure some adjusting and if someone finds how to open AO menus with LinkMessages, would be nice to add these.
